src/sagemaker/serve/utils/hardware_detector.py

```diff
@@ -48,55 +48,56 @@
 }
 
 
-def _get_gpu_info(instance_type: str, session: Session) -> int:
+def _get_gpu_info(instance_type: str, session: Session) -> tuple:
     """Get GPU info for the provided instance"""
     ec2_client = session.boto_session.client("ec2")
+    ec2_instance = _format_instance_type(instance_type)
+
+    try:
+        instance_info = ec2_client.describe_instance_types(InstanceTypes=[ec2_instance]).get(
+            "InstanceTypes"
+        )[0]
+    except ClientError as e:
+        logger.error("Failed to describe instance types %s", e)
+        raise ValueError("Provided instance_type is not GPU enabled.")
 
-    split_instance = instance_type.split(".")
-    split_instance.pop(0)
-
-    ec2_instance = ".".join(split_instance)
+    if instance_info is not None:
+        gpus_info = instance_info.get("GpuInfo")
+        if gpus_info is not None:
+            instance_gpu_info = (
+                gpus_info.get("Gpus")[0].get("Count"),
+                gpus_info.get("TotalGpuMemoryInMiB"),
+            )
 
-    instance_info = ec2_client.describe_instance_types(InstanceTypes=[ec2_instance]).get(
-        "InstanceTypes"
-    )[0]
+            logger.info("GPU Info [%s]: %s", ec2_instance, instance_gpu_info)
+            return instance_gpu_info
 
-    gpus_info = instance_info.get("GpuInfo")
-    if gpus_info:
-        return gpus_info.get("Gpus")[0].get("Count")
+    logger.error("Provided instance_type is not GPU enabled: [%s]", ec2_instance)
     raise ValueError("Provided instance_type is not GPU enabled.")
 
 
-def _aggregate_gpu_memory_size(instance_type: str, session: Session) -> int:
-    """Get Aggregate GPU Memory Size in MiB for the provided instance"""
-    ec2_client = session.boto_session.client("ec2")
-
-    split_instance = instance_type.split(".")
-    split_instance.pop(0)
-
-    ec2_instance = ".".join(split_instance)
+def _get_gpu_info_fallback(instance_type: str) -> tuple:
+    """Get GPU info for the provided instance fallback"""
+    fallback_instance_gpu_info = fallback_gpu_resource_mapping.get(instance_type)
```

Comment:
`fallback_gpu_resource_mapping` constants should be declared with capital letters